### ISSUE TITLE
Why does the comparison use "forked from default at" instead of "commit (SHA)" for labelling commit SHAs?

### ISSUE THREAD

**Issue by jonkeane** [conbench/conbench#682]
One example: https://conbench.ursa.dev/compare/benchmarks/a3faf3e6c7f048dab960f24f58618aa0...77aef10139db4f0a8376e1baa93f06b0/

<img width="1170" alt="image" src="https://user-images.githubusercontent.com/700357/216998811-93685316-0cb3-4173-b4cd-3c109796b7c0.png">

I suspect this was a mistake during the history renovation project — these commit SHAs appear to be the legit commit SHAs of the benchmark run itself (which is what we want here). So maybe this is just a label change?

**Comment by jgehrcke** [comment]
@jonkeane I am trying to understand the issue here. 

While doing so, I note some things that I find ... noteworthy :)

- The `commit` field shows a commit message prefix. It's also clearly a URL (leading to said commit details, I tried this out). But the actual _commit hash_ associated with the run is not shown explicitly in that table field. I think it should be shown.

- The `branch` field shows `apache:master`. That is an unusual notation to me. Should this be either `apache/arrow:master` or just `master`? If `apache:master` is expected I'd love to understand what this means.

- So, both commits seem to be on the default branch (`master`). In that sense, the "forked from default at" is surprising, because neither of both seems to be on a patch/PR branch. Is that what you mean @jonkeane? 

Update: I just confirmed that the linked commits (in the `commit` field) show commit has 32c... (left) and 007.. (right). What the "forked from ..." fields show. Is it true that this field "forked from ..." should simply _not_ appear in this table, or what would the expected outcome look like? Keep the table field but show "not applicable, because commit on master branch"? :)

**Comment by jonkeane** [comment]
> So, both commits seem to be on the default branch (master). In that sense, the "forked from default at" is surprising, because neither of both seems to be on a patch/PR branch. Is that what you mean @jonkeane?

Yeah, very surprising that there's any mention of forking since these are both commits on the default branch.

> Update: I just confirmed that the linked commits (in the commit field) show commit has 32c... (left) and 007.. (right). What the "forked from ..." fields show. Is it true that this field "forked from ..." should simply not appear in this table, or what would the expected outcome look like? Keep the table field but show "not applicable, because commit on master branch"? :)

Yeah, what I was expecting here (for commits to the default branch like this) is for the label "forked from default at" to instead be "commit (SHA)" but the commit SHA to still be in that field. The URL to the commit itself means that the commit sha is really great (and means that the commit sha is around if one looks for it), but I very much agree with you in this table it's good to show the commit sha explicitly (we're not too pressed for space, and folks will look for it since it's a nice rigid designator).

I haven't investigated this UI for branches (yet), for those I would expect both the commit sha (the literal commit on which the benchmarks were run), as well as the fork point sha (labelled with this label). In most common compare views, that would end up having the same SHA in "forked from default at" for the contender and "commit SHA" for the baseline (since our default compares always compare a PR-branch contender with the forked-from-default point)

**Comment by jgehrcke** [comment]
Thanks for the feedback, much appreciated!

**Comment by austin3dickey** [comment]
Right now, the "forked from default at" field always shows the `fork_point_sha` field of the commit in the database, since [this discussion](https://github.com/conbench/conbench/pull/417#discussion_r1009559148).

If the commit is on the default branch, the `fork_point_sha` is equal to the commit's sha itself. The reasons for that are [here](https://github.com/conbench/conbench/pull/431#discussion_r1021908358).

In an ideal world, I think I'd like to display the following fields. Thoughts?
- commit message with link to GitHub
- commit sha
- [only if not on default branch] fork point commit message with link to GitHub
- [only if not on default branch] fork point sha

>The `branch` field shows `apache:master`. That is an unusual notation to me. Should this be either `apache/arrow:master` or just `master`? If `apache:master` is expected I'd love to understand what this means.

The `org:branch` notation is required by the GitHub API as a fully-qualified way to identify a branch. (Here's [the docs](https://docs.github.com/en/pull-requests/committing-changes-to-your-project/viewing-and-comparing-commits/comparing-commits#comparing-across-forks).) That's why we store the name like that in the database. We could do some fancy string parsing between `commit.branch` and `commit.repository` to display a better string in this field, but I chose not to do that earlier.

**Comment by jonkeane** [comment]
> In an ideal world, I think I'd like to display the following fields. Thoughts?
> 
> * commit message with link to GitHub
> * commit sha
> * [only if not on default branch] fork point commit message with link to GitHub
> * [only if not on default branch] fork point sha

:100: I'm not sure the fork point commit message with link is _super_ critical, but if it's easy to construct go for it. 

> Right now, the "forked from default at" field always shows the fork_point_sha field of the commit in the database, since https://github.com/conbench/conbench/pull/417#discussion_r1009559148.

That's totally fine for the database, but like we've all been saying here: for the UI it looks really confusing + messy to surface that (but not the real commit sha) with little explanation that it's the same as the commit sha because of the git UI.
